soa-derive-internal/src/slice.rs

```diff
@@ -601,7 +601,7 @@ pub fn derive_mut(input: &Input) -> TokenStream {
             }
 
             #[doc(hidden)]
-            fn apply_permutation(&mut self, permutation: &mut soa_derive::Permutation) {
+            pub fn apply_permutation(&mut self, permutation: &mut soa_derive::Permutation) {
```

Comment:
this is not really intended to be public. I think I understand the issue, and marking this as pub seems like the only solution, but then I would rename the method to something like 

```suggestion
            /// this is `pub` due to the issue in https://github.com/lumol-org/soa-derive/pull/65
            pub fn __private_apply_permutation(&mut self, permutation: &mut soa_derive::Permutation) {
```